src/components/PostDetails.tsx

```diff
@@ -1,117 +1,148 @@
-import React from 'react';
+/* eslint-disable @typescript-eslint/no-shadow */
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { deletePostComment, getPostComments } from '../utils/api';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  post: Post,
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const { id, title, body } = post;
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [formOpened, setFormOpened] = useState(false);
+  const [loadProcessing, setLoadProcessing] = useState(false);
+  const [commentsError, setCommentsError] = useState(false);
+  const [commentDeletionError, setCommentDeletionError] = useState(false);
```

Comment:
redundant, you may have 1 state with string value (empty, so !!commentsError === false by default) and set different text in different cases